app/models/subnet/ipv4.rb

```diff
@@ -0,0 +1,38 @@
+require 'socket'
+
+class Subnet::Ipv4 < Subnet
+  has_many :interfaces, :class_name => 'Nic::Base', :foreign_key => :subnet_id
+  has_many :primary_interfaces, -> { where(:primary => true) }, :class_name => 'Nic::Base', :foreign_key => :subnet_id
+  has_many :hosts, :through => :interfaces
+  has_many :primary_hosts, :through => :primary_interfaces, :source => :host
```

Comment:
Interesting, I suppose it has something to do with class loading order. Just leave a comment so we know it and we're good!